### ISSUE TITLE
Possible Typo in Release Page

### ISSUE THREAD

**Issue by aroradaman** [kubernetes-sigs/kind#3380]
**What should be cleaned up or changed**:
The [v0.20.0](https://github.com/kubernetes-sigs/kind/releases/tag/v0.20.0) release information says :
```
The default node image is a Kubernetes v1.27.3 image: 

kindest/node:v1.27.2@sha256:3966ac761ae0136263ffdb6cfd4db23ef8a83cba8a463690e98317add2c9ba72
```

It talks about v1.27.3 but the tag is for v1.27.2. 
Also the SHA belongs v1.27.3 tag.

```
╭─heimdall@asgard ~ 
╰─$ crane digest kindest/node:v1.27.3
sha256:3966ac761ae0136263ffdb6cfd4db23ef8a83cba8a463690e98317add2c9ba72
╭─heimdall@asgard ~ 
╰─$ crane digest kindest/node:v1.27.2
sha256:ee77a85d1146ba4f1df9f836c828845a7dbe1f1a094ee670879d7c14f41e31f2

```
**Why is this needed**:

**Comment by aojea** [comment]
please, go ahead and  fix it please

**Comment by BenTheElder** [comment]
Yes that's a typo, sorry! FWIW docker, podman etc only use the digest anyhow so you should never trust the tag part of `:tag@digest` but it's still misleading to have the wrong version here.

**Comment by BenTheElder** [comment]
I corrected the typo, it said 1.27.3 in the full list and before the image there but was wrong in the first time the full image is listed.
